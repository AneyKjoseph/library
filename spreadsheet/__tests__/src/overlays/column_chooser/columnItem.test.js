import React from "react";
import ReactTestUtils, { act } from "react-dom/test-utils";
import ColumnItem from "../../../../src/overlays/column_chooser/columnItem";
import { DndProvider } from "react-dnd";
import { TouchBackend } from "react-dnd-touch-backend";




describe("<ColumnItem />", () => {
  const props = {
    id: 1,
    text: "text",
    moveColumn: jest.fn(),
    findColumn: jest.fn(() => 1),
    useDrop: jest.fn(),
  };
  it("mount", () => {
    const wrapper = mount(
      <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
        <ColumnItem {...props} />
      </DndProvider>
    );

    expect(wrapper.find("ColumnItem.useDrag.end")).not.toBeNull();
    expect(wrapper.find("ColumnItem.useDrop.canDrop")).not.toBeNull();
    expect(wrapper.find("div").length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
