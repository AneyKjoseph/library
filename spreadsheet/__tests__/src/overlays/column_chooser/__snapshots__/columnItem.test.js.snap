// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnItem /> mount 1`] = `ReactWrapper {}`;
